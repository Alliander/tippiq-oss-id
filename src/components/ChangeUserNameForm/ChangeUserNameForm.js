/**
 * Change user_name component.
 * @module components/ChangeUserNameForm
 */

import React, { PropTypes } from 'react';
import { reduxForm } from 'redux-form';
import { Row, Col } from 'react-bootstrap';

import { Field } from '../index';

/**
 * Change user_name form component class.
 * @function ChangeUserNameForm
 * @param {Object} props Component properties.
 * @param {Object} props.fields Fields for the form.
 * @param {Function} props.handleSubmit Submit form handler.
 * @param {Function} props.resetForm Reset form handler.
 * @returns {string} Markup of the not found page.
 */
export const ChangeUserNameForm = ({ fields, handleSubmit, errorMessage }) => {
  const error = errorMessage && <Col className="alert alert-danger" xs={12}>{errorMessage}</Col>;

  return (
    <Row>
      { error }
      <form onSubmit={handleSubmit}>
        <Col xs={12} sm={6} lg={5}>
          <Field field={fields.userName} placeholder="Jouw naam" type="text" />
        </Col>
        <Col xs={12} sm={6} lg={5}>
          <button
            id="userNameSubmitButton"
            type="submit"
            className="btn btn-primary btn-block"
            onClick={handleSubmit}
          >Opslaan
          </button>
        </Col>
      </form>
    </Row>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
ChangeUserNameForm.propTypes = {
  fields: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errorMessage: PropTypes.string,
};

export default reduxForm({
  form: 'changeUserName',
  fields: ['userName'],
})(ChangeUserNameForm);
