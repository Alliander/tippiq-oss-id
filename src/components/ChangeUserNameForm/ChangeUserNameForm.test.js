import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ChangeUserNameForm } from './ChangeUserNameForm';

test('Basic render snapshot test', () => {
  const component = shallow(<ChangeUserNameForm
    fields={{ userName: 'userName' }}
    handleSubmit={() => {}}
    errorMessage={'errormessage'}
  />);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

