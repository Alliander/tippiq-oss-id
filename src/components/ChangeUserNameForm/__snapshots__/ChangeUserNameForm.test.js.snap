// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    className="alert alert-danger"
    componentClass="div"
    xs={12}
  >
    errormessage
  </Col>
  <form
    onSubmit={[Function]}
  >
    <Col
      bsClass="col"
      componentClass="div"
      lg={5}
      sm={6}
      xs={12}
    >
      <Field
        field="userName"
        placeholder="Jouw naam"
        type="text"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      lg={5}
      sm={6}
      xs={12}
    >
      <button
        className="btn btn-primary btn-block"
        id="userNameSubmitButton"
        onClick={[Function]}
        type="submit"
      >
        Opslaan
      </button>
    </Col>
  </form>
</Row>
`;
